contracts/feature-tests/composability/interact/src/call_tree_calling_functions.rs

```diff
@@ -0,0 +1,150 @@
+                            println!("child_name: {}, parent_name: {}", child_fwd.name, &fwd_name);
+                            child_fwd.address.clone().unwrap()
+                        };
+
+                        let typed_sc_call = self
+                            .state
+                            .forwarder_queue_from_addr(&fwd_addr_expr)
+                            .add_queued_call(
+                                call_type.clone(),
+                                child_fwd_addr,
+                                DEFAULT_GAS_LIMIT,
+                                FORWARD_QUEUED_CALLS_ENDPOINT,
+                                MultiValueEncoded::<DebugApi, _>::new(),
+                            )
+                            .into_blockchain_call()
+                            .from(&self.wallet_address)
+                            .gas_limit("70,000,000")
+                            .expect(TxExpect::ok());
+
+                        steps.push(typed_sc_call);
+                    },
+                    CallNode::Vault(vault_rc) => {
+                        // Call Vault
+                        let vault_addr = {
+                            let vault = (*vault_rc).borrow();
+                            println!("child_name: {}, parent_name: {}", vault.name, &fwd_name);
+                            vault.address.clone().unwrap()
+                        };
+
+                        let typed_sc_call = self
+                            .state
+                            .forwarder_queue_from_addr(&fwd_addr_expr)
+                            .add_queued_call(
+                                call_type.clone(),
+                                vault_addr,
+                                DEFAULT_GAS_LIMIT,
+                                endpoint_name,
+                                MultiValueEncoded::<DebugApi, _>::new(),
+                            )
+                            .into_blockchain_call()
+                            .from(&self.wallet_address)
+                            .gas_limit("70,000,000")
+                            .expect(TxExpect::ok());
+
+                        steps.push(typed_sc_call);
+                    },
+                }
+            }
+        }
+        self.interactor
+            .multi_sc_exec(StepBuffer::from_sc_call_vec(&mut steps))
+            .await;
+
+        for step in steps.iter() {
+            let result = step.response().handle_signal_error_event();
+            if result.is_err() {
+                println!(
+                    "perform 'add_queued_call' failed with: {}",
+                    result.err().unwrap()
+                );
+                continue;
+            }
+            println!("successfully performed action 'add_queued_call'");
+        }
+    }
+
+    pub async fn call_root(
+        &mut self,
+        call_state: &CallState,
+        payment_token: EgldOrEsdtTokenIdentifier<DebugApi>,
```

Comment:
There is no need to transfer ESDT here, the contract will simply forward what it has accumulated